gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceManagerImpl.java

```diff
@@ -0,0 +1,235 @@
+
+  private static final Interceptor EXCEPTION_HANDLER_INTERCEPTOR = new Interceptor() {
+
+    private static final long serialVersionUID = 2091576149969931704L;
+
+    @Override
+    public RetryResult afterEval(Exception exception, RetryResult retryResult) {
+      return Interceptor.RetryResult.CONTINUE_EVALUATION;
+    }
+
+    @Override
+    public RetryResult beforeEval(Exception exception) {
+      if (exception instanceof ResourceManagerException) {
+        boolean retriable = ((ResourceManagerException) exception).retryable();
+        return retriable ? Interceptor.RetryResult.RETRY : Interceptor.RetryResult.NO_RETRY;
+      }
+      return Interceptor.RetryResult.CONTINUE_EVALUATION;
+    }
+  };
+  static final ExceptionHandler EXCEPTION_HANDLER = ExceptionHandler.builder()
+      .abortOn(RuntimeException.class)
+      .interceptor(EXCEPTION_HANDLER_INTERCEPTOR)
+      .build();
+
+  private final ResourceManagerRpc resourceManagerRpc;
+
+  ResourceManagerImpl(ResourceManagerOptions options) {
+    super(options);
+    resourceManagerRpc = options.rpc();
+  }
+
+  @Override
+  public ProjectInfo create(final ProjectInfo project) {
+    try {
+      return ProjectInfo.fromPb(runWithRetries(
+          new Callable<com.google.api.services.cloudresourcemanager.model.Project>() {
+            @Override
+            public com.google.api.services.cloudresourcemanager.model.Project call() {
+              return resourceManagerRpc.create(project.toPb());
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER));
+    } catch (RetryHelperException e) {
+      throw ResourceManagerException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public void delete(final String projectId) {
+    try {
+      runWithRetries(new Callable<Void>() {
+        @Override
+        public Void call() {
+          resourceManagerRpc.delete(projectId);
+          return null;
+        }
+      }, options().retryParams(), EXCEPTION_HANDLER);
+    } catch (RetryHelperException e) {
+      throw ResourceManagerException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public ProjectInfo get(final String projectId, ProjectGetOption... options) {
+    final Map<ResourceManagerRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.cloudresourcemanager.model.Project answer = runWithRetries(
+          new Callable<com.google.api.services.cloudresourcemanager.model.Project>() {
+            @Override
+            public com.google.api.services.cloudresourcemanager.model.Project call() {
+              return resourceManagerRpc.get(projectId, optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER);
+      return answer == null ? null : ProjectInfo.fromPb(answer);
+    } catch (RetryHelperException e) {
+      throw ResourceManagerException.translateAndThrow(e);
+    }
+  }
+
+  private abstract static class BasePageFetcher<T extends Serializable>
```

Comment:
This is a common base class for different services.

Not for this PR (so probably creating an issue) but we should move it to core and use Generics for the specific Options type `O` and provide an abstract method to return the `PAGE_TOKEN` option.